include/wil/registry_basic.h

```diff
@@ -0,0 +1,1540 @@
+//*********************************************************

```

Comment:
Should this be a separate file, or should it be auto-included in `registry.h`?

Reply:
I'm tempted right now to put the "supported functions" directly in registry.h, but have them call into registry_basic.h. Until we have everything we want, then we won't need registry_basic to 'hide' the details. (or maybe we will prefer it that way and just keep the extensive template goo in registry_basic.h and keep registry.h more readable / searchable).

in which case registry_basic.h might be better named registry_details.h ?

Reply:
We discussed offline, but I recommend simply using one file, `registry.h`. I'm not sure it really simplifies things to extract the functions. Frankly, I'd sooner move the registry watcher.